packages/blade/src/components/Tooltip/_decisions/decisions.md

```diff
@@ -0,0 +1,199 @@
+| ------------------ | ------------------------------------------------------------------------ | ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- |
+| content            | `string`                                                                 | `undefined` | Content of the tooltip                                                                                                                                   | ✅       |
+| children           | `React.ReactNode`                                                        | `undefined` | Trigger component for tooltip, Accepts any interactive element or icons                                                                                  | ✅       |
+| placement          | `top, top-start, top-end, left, right, bottom, bottom-start, bottom-end` | `top`       | Placement of tooltip                                                                                                                                     |          |
+| shouldWrapChildren | `boolean`                                                                | `false`     | If set to true, the tooltip trigger will be wrapped in a interactive `<div>`, this is useful for making non-interactive elements work as tooltip trigger |          |
+| onOpen             | `Callback`                                                               | `undefined` | Called when tooltip is opened                                                                                                                            |          |
+| onClose            | `Callback`                                                               | `undefined` | Called when tooltip is closed                                                                                                                            |          |
+
+### Examples:
+
+#### Usage
+
+```js
+// basic example
+<Tooltip content="Amount reversed to customer bank account" placement="top">
+  <IconButton icon={InfoIcon} accessibilityLabel="Refund" />
+</Tooltip>
+```
+
+### Working with non-interactive triggers
+
+To make tooltip apear on hovering over non-interactive elements such as icons, badges, counters etc. We will provide `shouldWrapChildren` prop which will wrap the trigger component in a interactive div element with all the trigger props
+
+```js
+// non-interactive element as trigger
+<Tooltip content="Amount reversed to customer bank account" shouldWrapChildren>
+  <Counter value={100} />
+</Tooltip>
+```
+
+> Note: If users don't wrap the non-interactive elements the tooltip won't apear
+
+### Custom Triggers
+
+Users will be able to use their own custom components as triggers:
+
+To make custom elements work the components needs to expose:
```

Comment:
whelp this sounds scary because so many things people will forget exposing one or the other thing. what if we provide some API like this
```
import type { TooltipTriggerProps } from "@razorpay/blade/components";

type MyCustomButtonProps = {} & TooltipTriggerProps
const MyCustomButton = ({children, ...props}) => <button {...props}>{children}</button>

// custom trigger
<Tooltip content="Amount reversed to customer bank account" shouldWrapChildren>
{(attributes) => <MyCustomButton {...attributes}>Click Me</MyCustomButton>}
</Tooltip>
```

might as well get rid of refs and all?

Reply:
I mean sure your API that you are showing will just work as is: 

Here: 

```jsx
import type { TooltipTriggerProps } from "@razorpay/blade/components";

type MyCustomButtonProps = {} & TooltipTriggerProps
// forward ref is needed
const MyCustomButton = React.forwardRef(({ children, ...spreadingPropsFineToo }, ref) => {
  return (
    <button ref={ref} {...spreadingPropsFineToo}>
      {children}
    </button>
  );
});

// custom trigger
<Tooltip content="Amount reversed to customer bank account">
  <MyCustomButton>Click Me</MyCustomButton> // no need to manually spread because we use React.cloneElement
</Tooltip>
```

Reply:
Yes, we are already passing that via React.cloneElement, users will need to spread the props tho.